ground/src/main/java/com/google/android/ground/ui/common/AbstractMapViewerFragment.kt

```diff
@@ -17,19 +17,108 @@ package com.google.android.ground.ui.common
 
 import android.os.Bundle
 import android.view.View
+import android.widget.Toast
+import androidx.navigation.fragment.NavHostFragment
 import com.google.android.ground.R
+import com.google.android.ground.repository.MapsRepository
+import com.google.android.ground.rx.RxAutoDispose
+import com.google.android.ground.system.PermissionDeniedException
+import com.google.android.ground.system.SettingsChangeRequestCanceled
+import com.google.android.ground.ui.home.mapcontainer.MapTypeDialogFragmentDirections
+import com.google.android.ground.ui.map.CameraPosition
 import com.google.android.ground.ui.map.MapFragment
 import javax.inject.Inject
+import kotlin.math.max
+import timber.log.Timber
 
-/** Injects a [MapFragment] in the container with id "map". */
+/** Injects a [MapFragment] in the container with id "map" and provides shared map functionality. */
```

Comment:
Thoughts on naming:

* Technically this is still a map container, so perhaps `AbstractMapContainerFragment` here?
* There will now be multiple map containers - should we rename `MapContainer*` to something unique, like `HomeScreenMapContainer` for consistency? Later we should find a better name for "home screen". @amegantz for discussion offline.

Reply:
Done. Thanks for the suggestion